extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -398,6 +419,18 @@ export class AmpAnalytics extends AMP.BaseElement {
     }
     const typeConfig = this.predefinedConfig_[type] || {};
 
+    // transport.iframe is only allowed to be specified in typeConfig, not
+    // the others. Allowed when running locally for testing purposes.
+    if (!getMode().localDev &&
```

Comment:
This feels a bit more readable but up to you:

   let hasFrame = false;
   [defaultConfig, inlineConfig, this.remoteConfig_].forEach(config => {
       if (config && config.transport && config.transport.iframe) {
          user().error('invalid config contains iframe transport', config);
          // WARN but allow in dev mode.
          hasFrame = getMode().localDev;
       }
   });
   if (hasFrame) {
     ... error and prevent execution
   }